enigma/src/main/java/org/quiltmc/enigma/api/source/Decompilers.java

```diff
@@ -1,13 +1,54 @@
 package org.quiltmc.enigma.api.source;
 
+import org.quiltmc.enigma.api.class_provider.ClassProvider;
 import org.quiltmc.enigma.impl.source.bytecode.BytecodeDecompiler;
 import org.quiltmc.enigma.impl.source.cfr.CfrDecompiler;
 import org.quiltmc.enigma.impl.source.procyon.ProcyonDecompiler;
 import org.quiltmc.enigma.impl.source.vineflower.VineflowerDecompiler;
 
 public class Decompilers {
-	public static final DecompilerService VINEFLOWER = VineflowerDecompiler::new;
-	public static final DecompilerService PROCYON = ProcyonDecompiler::new;
-	public static final DecompilerService CFR = CfrDecompiler::new;
-	public static final DecompilerService BYTECODE = BytecodeDecompiler::new;
+	public static final DecompilerService VINEFLOWER = new DecompilerService() {
```

Comment:
we could add a `DecompilerService#create(BiFunction<ClassProvider, SourceSettings, Decompiler>, String)` method